src/thor/isochrone_action.cc

```diff
@@ -13,12 +13,18 @@ std::string thor_worker_t::isochrones(Api& request) {
   auto costing = parse_costing(request);
   auto& options = *request.mutable_options();
 
-  std::vector<float> contours;
-  std::unordered_map<float, std::string> colors;
+  std::vector<float> contours_time, contours_distance;
+  std::unordered_map<const IsoMetrics, std::unordered_map<float, std::string>> colors;
+
   for (const auto& contour : options.contours()) {
-    contours.push_back(contour.time());
-    colors[contours.back()] = contour.color();
+    contours_time.push_back(contour.time());
+    colors[IsoMetrics::kTime][contour.time()] = contour.color();
+    contours_distance.push_back(contour.distance());
+    colors[IsoMetrics::kDistance][contour.distance()] = contour.color();
   }
+  // sort the contour vectors in descending order
+  std::sort(contours_time.begin(), contours_time.end(), std::greater<float>());
+  std::sort(contours_distance.begin(), contours_distance.end(), std::greater<float>());
```

Comment:
sorting the contour vectors here means we don't have to do it when validating the request params and we don't have to sort or check the sorting in other downstream functions